File: backend/emart-all/buyer-api/src/main/java/com/ibm/fsd/entity/PurchaseHistory.java
package com.ibm.fsd.entity;

import lombok.Data;
import lombok.NoArgsConstructor;

import javax.persistence.*;

import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import java.io.Serializable;
import java.util.Date;


@Data
@Entity
@NoArgsConstructor
@Table(name = "purchase_history")
public class PurchaseHistory implements Serializable {
	
    /**
	 * serialVersionUID
	 */
	private static final long serialVersionUID = 1L;

	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	@Column(name = "id")
	private int id;
	
	@Column(name = "product_code")
	private String productCode;
	
	@Column(name = "product_name")
	private String productName;
	
	@Column(name = "price")
	private double price;
	
	@Column(name = "quantity")
	private int quantity;
	
	@Column(name = "purchase_date")
	private String purchaseDate;
	
	@Column(name = "image_url")
	private String imageUrl;
	
	@Column(name = "user_id")
	private int userId;
	
	@CreationTimestamp
	@Column(name = "create_time")
    private Date createTime;
    
    @UpdateTimestamp
    @Column(name = "update_time")
    private Date updateTime;
	
}

File: frontend/emart-frontend/src/app/models/PurchaseHistory.ts
export class PurchaseHistory {
    id: number;
    productCode: String;
    productName: String;
    price: number;
    quantity: number;
    purchaseDate: String;
    imageUrl: String;
    userId: number;
    createTime: Date;
    updateTime: Date;
    
    constructor(purchaseHistory: PurchaseHistory, userId: number){
        this.productCode = purchaseHistory.productCode;
        this.productName = purchaseHistory.productName;
        this.price = purchaseHistory.price;
        this.quantity = purchaseHistory.quantity;
        this.purchaseDate = purchaseHistory.purchaseDate;
        this.imageUrl =  purchaseHistory.imageUrl;
        this.userId = userId;
    }
}
